dragonfly/block/farmland.go

```diff
@@ -0,0 +1,91 @@
+package block
+
+import (
+	"github.com/df-mc/dragonfly/dragonfly/internal/item_internal"
+	"github.com/df-mc/dragonfly/dragonfly/item"
+	"github.com/df-mc/dragonfly/dragonfly/world"
+	"math/rand"
+)
+
+// Farmland is a block that crops are grown on. Farmland is created by interacting with a grass block using a hoe.
+// Farmland takes into consideration its distance from a water source block to increase its efficiency and hold its hydration state
+type Farmland struct {
+	// Hydration is how much moisture a block has. The max and default hydration is 7, with its lowest
+	// hydration being 0, which is essentially dirt with a crop on it.
+	// This is calculated by checking if there is a water source block 4 blocks away in any direction from
+	// the farmland that is either 1 block above or on the same level as the farmland block.
+	// If there isn't, we then count down the hydration level by one until it eventually dries up and turns back into dirt.
+	Hydration uint8
```

Comment:
Would change this to an `int` for the sake of simplicity, then convert it to a uint8 when encoding.